app/models/comment.rb

```diff
@@ -0,0 +1,10 @@
+# Model for comments
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  belongs_to :posts, class_name: 'Post', foreign_key: :create_posts_id
+  has_many :comments, foreign_key: :create_posts_id

```

Comment:
- [ ] Same here, the class_name suppose to be `User` instead of `CreateUser` and `Post` instead of `CreatePost` (Or remove it and the naming conventions will do it for you).
- [ ] Please remove this `has_many` relation because it makes no sense, how the comments will have many comments.
You can do it like that:
```ruby
belongs_to :author, class_name: 'User'
belongs_to :post
```

Reply:
ok fixed